semgrep/semgrep/git.py

```diff
@@ -108,27 +108,44 @@ def _get_git_status(self) -> GitStatus:
 
         # Output of git command will be relative to git project root not cwd
         logger.debug("Running git diff")
-        status_output = zsplit(
-            subprocess.run(
-                [
-                    "git",
-                    "diff",
-                    "--cached",
-                    "--name-status",
-                    "--no-ext-diff",
-                    "-z",
-                    "--diff-filter=ACDMRTUXB",
-                    "--ignore-submodules",
-                    "--relative",
-                    "--merge-base",
-                    f"{self._base_commit}",
-                ],
+        status_cmd = [
+            "git",
+            "diff",
+            "--cached",
+            "--name-status",
+            "--no-ext-diff",
+            "-z",
+            "--diff-filter=ACDMRTUXB",
+            "--ignore-submodules",
+            "--relative",
+            self._base_commit,
+        ]
+        try:
+            # nosemgrep: python.lang.security.audit.dangerous-subprocess-use.dangerous-subprocess-use
+            raw_output = subprocess.run(
+                [*status_cmd, "--merge-base"],
                 timeout=GIT_SH_TIMEOUT,
                 capture_output=True,
                 encoding="utf-8",
                 check=True,
             ).stdout
-        )
+
+        except subprocess.CalledProcessError as exc:
```

Comment:
Seems like this would only fix cases when we hit this exception of multiple merge bases but findings appearing would mean that the former command exited with exit code zero right?